### PR TITLE
test(storage): support testing with alternative endpoints

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14502)

### CODE REVIEW

File: google/cloud/storage/testing/storage_integration_test.cc [hunk 1]
```diff
@@ -61,12 +59,29 @@ StorageIntegrationTest::~StorageIntegrationTest() {
   }
 }
 
+google::cloud::Options StorageIntegrationTest::MakeTestOptions(
+    google::cloud::Options opts) {
```
Comment (dbolduc) [inline google/cloud/storage/testing/storage_integration_test.cc:63]: nit: I would drop `google::cloud::` here and below

File: google/cloud/storage/testing/storage_integration_test.h [hunk 2]
```diff
@@ -39,6 +39,14 @@ class StorageIntegrationTest
  protected:
   ~StorageIntegrationTest() override;
 
+  /**
+   * Returns the recommended options to run integration tests.
+   *
+   * Most tests should use these options or call `MakeIntegrationTestClient()`.
+   */
+  static google::cloud::Options MakeTestOptions(
+      google::cloud::Options opts = {});
```
Comment (dbolduc) [inline google/cloud/storage/testing/storage_integration_test.h:48]: same

**Review by dbolduc — APPROVED** [review]

File: google/cloud/storage/testing/storage_integration_test.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/testing/storage_integration_test.cc]: Done.

File: google/cloud/storage/testing/storage_integration_test.h (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/testing/storage_integration_test.h]: Done

**Pull request merged**